### ISSUE TITLE
Module '".../node_modules/@types/lodash"' has no exported member 'Cancelable'. 

### ISSUE THREAD

**Issue by LAlves91** [DefinitelyTyped/DefinitelyTyped#47166]
Greetings!

I work with a few Angular projects. One of them has a dependency on a library 'angular-tree-component', which depends on '@types/lodash'.

A few days ago, pull request [46971](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46971) was merged, generating version 4.14.161. As 'angular-tree-component' declares '^4.14.62' as the required version, the library was updated, generating the following error:

`
ERROR in node_modules/angular-tree-component/dist/components/tree-viewport.component.d.ts(3,10): error TS2305: Module '"MY_PATH/node_modules/@types/lodash"' has no exported member 'Cancelable'.
`

My question is: was it an implementation error from 'angular-tree-component' to use Cancelable as a return type, or should this interface be kept in '@types/lodash' for compatibility purposes?

Thank you!

**Comment by Luis16287** [comment]
I have exactly the same problem. I am using [angular-tree-component : 8.0.0](https://angular2-tree.readme.io/docs/getting-started) in a project, when compiling it gives me the following error:

`node_modules / angular-tree-component / dist / components / tree-viewport.component.d.ts: 3:10 - error TS2305: Module '"MyPath / node_modules / @ types / lodash"' has no exported member 'Cancelable'.`

What should I do?

I have the same doubt as @LAlves91 

The configuration of my project is as follows:

- `Angular CLI: 8.0.0`
- `Node: 10.13.0`
- `npm: 6.4.1`
- `rxjs: 6.4.0`
- `typescript: 3.4.5`
- `webpack: 4.39.2`
- `@types/lodash": "~4.14.118`
- `lodash": "4.17.12`
- `angular-tree-component": "8.0.0  // for Angular 9 or higher you should switch to the @circlon/angular-tree-component`

**Comment by LAlves91** [comment]
Hey @Luis16287 !

For now, I'm fixing @types/lodash version at 4.14.160 in my project's package.json. At least it started building again.

**Comment by miwels** [comment]
4.14.161 introduces a breaking change. Looks like the Cancellable interface has been removed and replaced with DebouncedFunc. Check this commit:

https://github.com/DefinitelyTyped/DefinitelyTyped/commit/93d8f9d7102925e1a39852627465a44a0ff670cf

**Comment by idanwork** [comment]
Thanks @miwels for pinpointing the change, that did the trick.

Here's an example change

Changed
```typescript
debouncedMethod: ((x: any) => void) & Cancelable;
```

to 
```typescript
debouncedMethod: DebouncedFunc<(x: any) => void>;
```

And also the import to 
```typescript
import { DebouncedFunc, ...more.... } from "lodash";
```

**Comment by pedro-surf** [comment]
Same error showed up here, rebuilding my project locally when I lost cache.

Somehow I got rid of it but now I have this:

```
node_modules/@types/node/index.d.ts(20,1): error TS1084: Invalid 'reference' directive syntax.
The command '/bin/sh -c tsc -p .' returned a non-zero code: 2
```
I'm running on Docker and nothing has changed in the repository.
On my yarn.lock files some deps do have 
```

  dependencies:
    "@types/node" "*"
```

Could anyone help? Thanks!

**Comment by Luis16287** [comment]
@LAlves91 

Sorry for the delay, for the particular case of the angular project using [angular-tree-component,](https://github.com/CirclonGroup/angular-tree-component) I got the following response 

"**This issue is due to @ types / lodash being a dependency in version 8 up to 8.2.0. This should have been fixed with 8.2 .1. I'm not sure what else was done and how risky / time-consuming the switch from 8.0.0 to 8.2.1 is. I checked and we used version 8.3.0 together with Angular 8.0.0 and version 8.4. 0 of the tree with Angular 8.2.0.
Do you want a 8.0.6 with this fix as well? One problem is that we can not publish to the old npm name. We would have to publish this under @ circlon / angular-tree-component. This probably would also mean republishing everything up to 8.5.6 under the new name so you don't need to switch the package names updating to another 8.x.x version.
Is this something that's wanted / needed? How many can not update the tree to a newer version?**"

## So I tried the following configuration

- `Angular CLI: 8.0.0`
- `Node: 10.13.0`
- `npm: 6.4.1`
- `rxjs: 6.4.0`
- `typescript: 3.4.5`
- `webpack: 4.39.2`
- `lodash": "4.17.12`
**with these changes**
- `@types/lodash": "4.14.160`
- `angular-tree-component": "8.3.0 // for Angular 9 or higher you should switch to the @circlon/angular-tree-component`

And it has worked correctly for me

If you want to follow the thread is [here (angular-tree-component#859)](https://github.com/CirclonGroup/angular-tree-component/issues/859)

**Comment by orta** [comment]
Hi thread, we're moving DefinitelyTyped to use [GitHub Discussions](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53377) for conversations the `@types` modules in DefinitelyTyped.

To help with the transition, we're closing all issues which haven't had activity in the last 6 months, which includes this issue. If you think closing this issue is a mistake, please pop into the [TypeScript Community Discord](https://discord.gg/typescript) and mention the issue in the `definitely-typed` channel.
